hooks/mari/publish_mipmaps.py

```diff
@@ -91,82 +108,195 @@ def validate(self, task_settings, item):
         :param item: Item to process
         :returns: True if item is valid, False otherwise.
         """
+        # TODO: check if Publish Textures plugin was accepted and validated?
+        # heavily dependent on that output
 
-        geo_name = item.properties.mari_geo_name
-        geo = mari.geo.find(geo_name)
-        if not geo:
-            error_msg = "Failed to find geometry in the project! Validation failed." % geo_name
-            self.logger.error(error_msg)
+        if not super(MariPublishMipmapsPlugin, self).validate(task_settings, item):
             return False
 
-        channel_name = item.properties.mari_channel_name
-        channel = geo.findChannel(channel_name)
-        if not channel:
-            error_msg = "Failed to find channel on geometry! Validation failed." % channel_name
-            self.logger.error(error_msg)
+        if not self._validate_getter_attrs(item, task_settings, ["mipmap_extensions"]):
             return False
 
-        layer_name = item.properties.get("mari_layer_name")
-        if layer_name:
-            layer = channel.findLayer(layer_name)
-            if not layer:
-                error_msg = "Failed to find layer for channel: %s Validation failed." % layer_name
+        for extension in item.properties["mipmap_extensions"]:
+            publish_path = sgtk.util.ShotgunPath.normalize(
+                self._get_publish_path_with_ext(item, task_settings, extension))
+
+            if not self._valid_for_mipmap_multiimage(publish_path):
+                error_msg = "Failed to find OIIO plugin or mipmap not supported for extension: %s " \
+                            "Validation failed." % os.path.splitext(publish_path)[-1]
                 self.logger.error(error_msg)
                 return False
 
-        if not super(MariPublishMipmapsPlugin, self).validate(task_settings, item):
-            return False
-
-        publish_path = sgtk.util.ShotgunPath.normalize(item.properties.get("publish_path"))
-        if not self._valid_for_mipmap_multiimage(publish_path):
-            error_msg = "Failed to find OIIO plugin or mipmap not supported for extension: %s " \
-                        "Validation failed." % os.path.splitext(publish_path)[-1]
-            self.logger.error(error_msg)
-            return False
-
         return True
 
 
-    def publish_files(self, task_settings, item, publish_path):
+    def publish_files(self, task_settings, item):
         """
         Overrides the inherited method to export out session items to the publish_path location.
 
         :param task_settings: Dictionary of Settings. The keys are strings, matching
             the keys returned in the settings property. The values are `Setting`
             instances.
         :param item: Item to process
-        :param publish_path: The output path to publish files to
         """
         publisher = self.parent
+        item.properties["mipmap_paths"] = []
```

Comment:
item property so that it can be used in undo without method signature being changed.